gradle-plugin/src/main/groovy/io/realm/gradle/RealmTransformer.groovy

```diff
@@ -0,0 +1,130 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.gradle
+
+import com.android.build.api.transform.*
+import com.google.common.collect.BiMap
+import com.google.common.collect.HashBiMap
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.Sets
+import groovy.io.FileType
+import io.realm.annotations.Ignore
+import javassist.ClassPool
+import javassist.LoaderClassPath
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import static com.android.build.api.transform.QualifiedContent.*
+
+class RealmTransformer extends Transform {
+
+    private Logger logger = LoggerFactory.getLogger('realm-logger')
+
+    @Override
+    String getName() {
+        return "RealmTransformer"
+    }
+
+    @Override
+    Set<ContentType> getInputTypes() {
+        return ImmutableSet.<ContentType> of(DefaultContentType.CLASSES)
+    }
+
+    @Override
+    Set<Scope> getScopes() {
+        return Sets.immutableEnumSet(
+                Scope.PROJECT)
+    }
+
+    @Override
+    boolean isIncremental() {
+        return true
+    }
+
+    @Override
+    void transform(Context context, Collection<TransformInput> inputs, Collection<TransformInput> referencedInputs,
+                   TransformOutputProvider outputProvider, boolean isIncremental)
+            throws IOException, TransformException, InterruptedException {
+
+        final ArrayList<File> folders = []
+        inputs.each { TransformInput input ->
+            logger.info "Directory inputs: ${input.directoryInputs*.file}"
+            folders.addAll(input.directoryInputs*.file)
+        }
+
+        referencedInputs.each { TransformInput input ->
+            folders.addAll(input.directoryInputs*.file)
+            folders.addAll(input.jarInputs*.file)
+        }
+
+
+        BiMap<File, String> classFiles = HashBiMap.create()
+
+        folders.each { File folder ->
+            folder.eachFileRecurse(FileType.FILES) { File file ->
+                if (file.name.endsWith('.class')) {
+                    classFiles.put(
+                            file,
+                            file.canonicalPath
+                                    .replace(folder.canonicalPath, '')
+                                    .replaceFirst('/', '')
+                                    .replace('/', '.')
+                                    .replace('.class', ''))
+                }
+            }
+        }
+
+        // Don't use ClassPool.getDefault(). Doing so consecutive builds in the same run (e.g. debug+release)
+        // will use a cached object and all the classes will result frozen.
```

Comment:
It's Javassist lingo. See here https://jboss-javassist.github.io/javassist/tutorial/tutorial.html in the "frozen classes" section